### PR TITLE
fix bad data persistance in form

### PR DESCRIPTION
**Pull request by CFarcy**

### CODE REVIEW

File: src/modules/RA/DataLayer/components/tabs/PopupTab/PopupFieldList.js [hunk 1]
```diff
@@ -69,21 +66,27 @@ const PopupFieldList = ({ fields, updateTemplate }) => {
...
+    updateTemplate();
+  }, [popupFields, updateTemplate]);
+
   return (
     <div className="wrapper" style={{ width: '50%' }}>
       <h4>{translate('datalayer.form.popup.title-field')}</h4>
       <Field
         name="popup_config.wizard.fields[0]"
         validate={validateTitle}
+        initialValue={useMemo(() => ({ sourceFieldId: mainFieldId }), [mainFieldId])}
```
Comment (jrmi) [inline src/modules/RA/DataLayer/components/tabs/PopupTab/PopupFieldList.js:84]: I don't think using a useMemo here is a good idea. Not really common nor recommended.

File: src/modules/RA/DataLayer/components/tabs/PopupTab/PopupFieldList.js [hunk 2]
```diff
@@ -69,21 +66,27 @@ const PopupFieldList = ({ fields, updateTemplate }) => {
     return errors;
   }, [translate]);
 
-  useEffect(() => {
+  const onTitleFieldChange = useCallback(cb => value => {
```
Comment (jrmi) [inline src/modules/RA/DataLayer/components/tabs/PopupTab/PopupFieldList.js:69]: Not working ;-) You should create a component for this...

File: src/modules/RA/DataLayer/components/tabs/PopupTab/PopupFieldList.js (hunk 1 above)
Comment (CFarcy) [reply src/modules/RA/DataLayer/components/tabs/PopupTab/PopupFieldList.js:84]: Ok, I'll remove it

Comment (CFarcy) [reply src/modules/RA/DataLayer/components/tabs/PopupTab/PopupFieldList.js]: I move it and assigned it to a variable in the component

File: src/modules/RA/DataLayer/components/tabs/PopupTab/PopupFieldList.js (hunk 2 above)
Comment (CFarcy) [reply src/modules/RA/DataLayer/components/tabs/PopupTab/PopupFieldList.js]: Done.

**Review by jrmi — APPROVED** [review]

**Pull request merged**